### PR TITLE
ui/emojis: New module for inserting/displaying emojis

### PR DESCRIPTION
**Pull request by akirabaruah**
Uses the emojify package to enable displaying and inserting emojis. Also adds a new binding "SPC i e" to insert an Emoji.

### CODE REVIEW

**Comment by AlexAnarcho** [comment]
Would be a nice feature, i m struggeling with emojis in doom emacs

File: modules/ui/emojis/config.el [hunk 1]
```diff
@@ -0,0 +1,13 @@
+;;; ui/emojis/config.el -*- lexical-binding: t; -*-
+
+(setq emojify-styles
+      (seq-filter #'identity
+                  (list (if (featurep! +ascii) 'ascii)
+                        (if (featurep! +github) 'github)
+                        (if (featurep! +unicode) 'unicode))))
+
+(use-package! emojify
+  :defer t
```
Comment (hlissner) [inline modules/ui/emojis/config.el:10]: Remove `:defer t`. It's redundant with `:hook`.

File: modules/ui/emojis/config.el [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  :hook (after-init . global-emojify-mode)
```
Comment (hlissner) [inline modules/ui/emojis/config.el:11]: Can this be deferred later? Perhaps until `doom-first-buffer-hook`? Or at most `doom-first-input-hook`?

File: modules/ui/emojis/config.el [hunk 3]
```diff
@@ -0,0 +1,13 @@
+;;; ui/emojis/config.el -*- lexical-binding: t; -*-
+
+(setq emojify-styles
+      (seq-filter #'identity
+                  (list (if (featurep! +ascii) 'ascii)
+                        (if (featurep! +github) 'github)
+                        (if (featurep! +unicode) 'unicode))))
```
Comment (hlissner) [inline modules/ui/emojis/config.el:7]: This is emojify configuration, it should be in a `:config` of the `emojify` use-package block.

File: modules/ui/emojis/config.el [hunk 4]
```diff
@@ -0,0 +1,13 @@
+;;; ui/emojis/config.el -*- lexical-binding: t; -*-
+
+(setq emojify-styles
+      (seq-filter #'identity
```
Comment (hlissner) [inline modules/ui/emojis/config.el:4]: Use `cl-remove-if-not`. Doom loads `cl-lib` by default, but not `seq` (at least, not in interactive sessions), so there is no hard guarantee `seq-filter` will be available.

File: modules/ui/emojis/packages.el [hunk 5]
```diff
@@ -0,0 +1,6 @@
+;; -*- no-byte-compile: t; -*-
+;;; ui/emojis/packages.el
+
+(package! emojify
+  :recipe (:host github :repo "iqbalansari/emacs-emojify")
```
Comment (hlissner) [inline modules/ui/emojis/packages.el:5]: Why the explicit recipe? `emojify` appears to be on MELPA and it appears to be from the same repo.

**Review by hlissner — CHANGES_REQUESTED** [review]

File: modules/ui/emojis/config.el (hunk 2 above)
Comment (akirabaruah) [reply modules/ui/emojis/config.el:11]: Yep, `doom-first-buffer-hook` seems more appropriate.

File: modules/ui/emojis/packages.el (hunk 5 above)
Comment (akirabaruah) [reply modules/ui/emojis/packages.el]: No reason besides ignorance - thanks for pointing it out!

File: modules/ui/emojis/config.el (hunk 4 above)
Comment (akirabaruah) [reply modules/ui/emojis/config.el]: Good to know. Fixed!

File: modules/ui/emojis/config.el (hunk 3 above)
Comment (akirabaruah) [reply modules/ui/emojis/config.el]: Done!

File: modules/ui/emojis/config.el (hunk 1 above)
Comment (akirabaruah) [reply modules/ui/emojis/config.el]: Done!

**Review by hlissner — APPROVED** [review]

File: modules/ui/emojis/config.el [hunk 6]
```diff
@@ -0,0 +1,11 @@
+;;; ui/emojis/config.el -*- lexical-binding: t; -*-
+
+(use-package! emojify
+  :hook (doom-first-buffer . global-emojify-mode)
+  :config
+  (setq emojify-styles
+        (cl-remove-if-not #'identity
+                          (list (if (featurep! +ascii) 'ascii)
+                                (if (featurep! +github) 'github)
+                                (if (featurep! +unicode) 'unicode))))
```
Comment (hlissner) [inline modules/ui/emojis/config.el:10]: Actually, could you replace this with:

```elisp
(setq emojify-styles
      (delq
       nil (list (if (featurep! +ascii) 'ascii)
                 (if (featurep! +github) 'github)
                 (if (featurep! +unicode) 'unicode))))
```

**Review by hlissner — CHANGES_REQUESTED** [review]
Looks good! Two things occurred to me:

1. Could you add an entry for this module in docs/modules.org. e.g.

   ```diff
    + [[file:../modules/ui/doom/README.org][doom]] - TODO
    + [[file:../modules/ui/doom-dashboard/README.org][doom-dashboard]] - TODO
    + [[file:../modules/ui/doom-quit/README.org][doom-quit]] - TODO
   ++ emoji =+ascii +github +unicode= - Adds emoji support to Emacs
    + fill-column - TODO
    + [[file:../modules/ui/hl-todo/README.org][hl-todo]] - TODO
    + [[file:../modules/ui/hydra/README.org][hydra]] - TODO
   ```

2. Could you rename the module `:ui emoji`? There's no consensus on whether emoji plural is "emoji" or "emojis", but for the sake of keystrokes I prefer the former.

File: modules/ui/emojis/config.el (hunk 6 above)
Comment (akirabaruah) [reply modules/ui/emojis/config.el]: Ooh that's nice and concise. Done!

**Comment by akirabaruah** [comment]
1. Added `emoji` entry in `docs/modules.org`
2. Renamed module to `:ui emoji` :+1: yeah plural's awkward :sweat_smile: I speak Japanese (where "emoji" is both singular/plural), but I still find myself saying "emojis" in English :shrug:

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Looks good to me. Thanks for your help!

**Comment by agsdot** [comment]
This is a pretty awesome feature add to Doom. Really appreciate this PR coming in.

**Comment by akirabaruah** [comment]
@agsdot glad you found it useful 😄 

Real hats off to @iqbalansari (author of the package) and @hlissner for this awesome project!

**Comment by agsdot** [comment]
Yeah, props to all of you though @akirabaruah , @hlissner and @iqbalansari. 

Before this package to generate emoji's I had to either be on my mobile device and keyboard or on a desktop and use https://emojipedia.org/. This package and nice keybinding of SPC i e, lends a very nice and rapid way to find emoji when typing.

Thanks once again!

**Pull request merged**